modules/programs/ncmpcpp.nix

```diff
@@ -0,0 +1,142 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib) types mkIf mkEnableOption mkOption;
+
+  cfg = config.programs.ncmpcpp;
+
+  mpdCfg = config.services.mpd;
+
+  mpdMusicDir = if cfg.mpdMusicDir == null then
+    mpdCfg.musicDirectory
+  else
+    toString cfg.mpdMusicDir;
+
+  renderSettings = settings:
+    lib.concatStringsSep "\n" (lib.mapAttrsToList renderSetting settings);
+
+  renderSetting = name: value: "${name}=${renderValue value}";
+
+  renderValue = option:
+    rec {
+      int = toString option;
+      bool = if option then "yes" else "no";
+      string = option;
+    }.${builtins.typeOf option};
+
+  renderBindings = bindings:
+    lib.concatStringsSep "\n" (map renderBinding bindings);
+
+  renderBinding = { key, command }:
+    lib.concatStringsSep "\n  "
+    ([ ''def_key "${key}"'' ] ++ maybeWrapList command);
+
+  maybeWrapList = xs: if lib.isList xs then xs else [ xs ];
+
+  valueType = types.oneOf [ types.bool types.int types.str ];
+
+  bindingType = types.submodule ({ name, config, ... }: {
+    options = {
+      key = mkOption {
+        type = types.str;
+        description = "Key to bind.";
+        example = "j";
+      };
+
+      command = mkOption {
+        type = types.either types.str (types.listOf types.str);
+        description = "Command or sequence of commands to be executed.";
+        example = "scroll_down";
+      };
+    };
+  });
+
+in {
+  meta.maintainers = with lib.maintainers; [ olmokramer ];
+
+  options.programs.ncmpcpp = {
+    enable =
+      mkEnableOption "ncmpcpp - An ncurses Music Player Daemon (MPD) client";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.ncmpcpp;
+      defaultText = "pkgs.ncmpcpp";
+      description = "Package providing the <code>ncmpcpp</code> command.";
+      example = "pkgs.ncmpcpp.override { ... }";
+    };
+
+    mpdMusicDir = mkOption {
+      type = types.nullOr types.path;
+      default = null;
```

Comment:
Could use
```nix
if mpdCfg.enable then mpdCfg.musicDirectory else null
```
as the default? If so, then you'll also need to add a `defaultText` field. Something like
```
defaultText = literalExample ''
  if config.services.mpd.enable then
    config.services.mpd.musicDirectory
  else
    null
'';
```
perhaps.

Reply:
Yeah that's a great idea!